src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java

```diff
@@ -131,15 +133,31 @@ private void initializeHandlers() {
     }
 
     private void subscribeToConfigChanges() {
+        // This first call is just to initialize an empty configuration
         onConfigurationChanged();
         config.lookupTopics(CONFIGURATION_CONFIG_KEY).subscribe(this::configChangeHandler);
     }
 
     private void onConfigurationChanged() {
-        try {
-            cdaConfiguration = CDAConfiguration.from(cdaConfiguration, getConfig());
-        } catch (URISyntaxException e) {
-            serviceErrored(e);
+        Result<CDAConfiguration> result = CDAConfiguration.from(cdaConfiguration, getConfig());
+        CDAConfiguration configuration = result.get();
+
+        // Nothing has changed
+        if (configuration.isEqual(cdaConfiguration)) {
+            return;
+        }
+
+        cdaConfiguration = configuration;
+
+        if (result.isOk() && inState(State.BROKEN)) {
+            logger.info("Service is {} and configuration changed. Attempting to reinstall.", State.BROKEN);
+            requestReinstall();
+            return;
```

Comment:
I may have misunderstood it, but it seems like we're _silently_ reinstalling the service when bad configuration. Customer may not look at this log when we silently rollback the service. Should we just keep the service in broken state and fail the deployment?

Reply:
We are not silently reinstalling. Basically every time the customer provides a wrong configuration we will call `serviceErrored` (on the section below). If they keep providing bad configuration values we will keep erroring (which will show the state and errors on the logs). If they provide a bad configuration enough times, we will eventually go into a BROKEN state.

The only way of exiting out of a BROKEN state is re installing the component. Hence, if we are in a broken state and the customer does a deployment with a new working config we basically tell it to reinstall. Without this step, (and this is also how the current version of CDA in production works), if the component goes into a broken state, deploying a configuration change won't simply fix it, the customers would have to manually restart the component.